addons/web/static/src/js/chrome/web_client.js

```diff
@@ -127,42 +127,40 @@ return AbstractWebClient.extend({
     // --------------------------------------------------------------
     // URL state handling
     // --------------------------------------------------------------
-    on_hashchange: function (event) {
+    on_hashchange: async function (event) {
         if (this._ignore_hashchange) {
             this._ignore_hashchange = false;
             return Promise.resolve();
         }
 
         var self = this;
-        return Promise.resolve().then(function () {
-            var stringstate = $.bbq.getState(false);
-            if (!_.isEqual(self._current_state, stringstate)) {
-                var state = $.bbq.getState(true);
-                if (state.action || (state.model && (state.view_type || state.id))) {
-                    return self.menu_dp.add(self.action_manager.loadState(state, !!self._current_state)).then(function () {
-                        if (state.menu_id) {
-                            if (state.menu_id !== self.menu.current_primary_menu) {
-                                core.bus.trigger('change_menu_section', state.menu_id);
-                            }
-                        } else {
-                            var action = self.action_manager.getCurrentAction();
-                            if (action) {
-                                var menu_id = self.menu.action_id_to_primary_menu_id(action.id);
-                                core.bus.trigger('change_menu_section', menu_id);
-                            }
+        var stringstate = $.bbq.getState(false);
```

Comment:
maybe keep the code indented so reduce the diff here (it's a scary part :-), and we re-writing it anyway)